dotnet/src/webdriver/WebDriver.cs

```diff
@@ -455,6 +455,17 @@ public INavigation Navigate()
             return new Navigator(this);
         }
 
+        /// <summary>
+        /// Provides access to WebDriverBiDi events in script and log domains.
+        /// </summary>
+        /// <returns>An <see cref="IScript"/> object allowing the user to access
+        /// WebDriverBiDi events.</returns>
+        [Obsolete("This method is in beta and may change in future releases.")]
+        public IScript Script()
```

Comment:
This is the kind of thing I wish I'd changed in the initial creation of the .NET bindings rather than just blindly copying Java. I would have made these properties instead of methods. We can maintain consistency or try to move to idiomatic .NET code.

Reply:
Maybe a dumb question. If we're going to ask people to update to the Async methods anyway, would it make sense to have users access those methods via a property? (e.g., we tell people to change from `driver.Navigate().Back()` to `driver.Navigate.BackAsync()`)

Reply:
Bingo. I was always thinking about why `driver.Navigate().Back()` when `driver.Back()` is simpler.

My personal opinion is very simple: introduce new API besides old. That's it. Just open a door to new API for users: `var bidiDriver = await driver.AsBiDiAsync()`.

Reply:
but we don't want people to spend time trying to figure out if they want a driver or a bidiDriver. Whether it is classic or BiDi doesn't matter to the user, it only matters if it is sync or async. So let's make async use the latest whatever it is and people only have to know driver.

The reason not to put *everything* on `driver` is that it makes discoverability harder. In the IDE you want to click the `.` and see what all can be done, and you don't want it to be too much. We will definitely disagree on what is or isn't obvious, but putting navigation stuff in a navigation namespace seems straightforward.

Reply:
Ok, I clicked the `.` and I see:
- Close()
- CurrentWindowHandle
- Dispose()
- FindElement()
- FindElements()
- Manage()
- Navigate()
- PageSource
- Quit()
- SwitchTo()
- Title
- Url
- WindowHandles

Is it a lot?!

When I type primitive `int.` - I see even more, even more for primitive type.

Reply:
I don't think it's too much right now, but I'd rather only add a few more methods than dozens

Reply:
If you really want to namespace methods for the driver object, than `driver.Navigate.BackAsync()` doesn't look good. `Navigate` is a verb if I am not wrong, so it should look like a method. And `Navigate` looks like a state of the object, what is incorrect.
If I would really want to split methods by "spaces", I would choose `driver.AsNavigatable().BackAsync()`.

Reply:
For `Back()` it makes sense to be a verb for it to sound fluent: "Navigate Back," But then that breaks with "Navigate Go To Url" instead of just "Navigate To Url"

In general, though, we need to be very cautious about making major changes to the existing API. We want as small a lift as possible for people to update their code. Changing from a method to a property seems straightforward, but changing the name entirely is a lot more intrusive.

Reply:
Then do not change it at all.

Reply:
That's definitely the safest option